### ISSUE TITLE
FWmaCARMAnudged.f09_f09_mg17.carma_trop_strat06.2021_2022

### ISSUE THREAD

**Issue by tilmes** [NCAR/CARMA_dev#10]
WACCM-MA CARMA nudge run 1deg 2021-2022 for Hunga-Tonga / ACCLIP comparisons  (for @zywshoon)
Code base: /glade/work/fvitt/camdev/carma_trop_strat06
Case directory: /glade/p/cesm/chwg_dev/tilmes/cases/carma/FWmaCARMAnudged.f09_f09_mg17.carma_trop_strat06.2021_2022
First test: IC taken from 2deg WACCM-MA case
Emissions: Hunga Tonga SO2 and H2O emissions
Qfed and CAM5.1 emissions

**Comment by tilmes** [comment]
@zywshoon For the Hunga-Tonga run with MAM4 we add vertical Black Carbon emissions. I don't think, we have implemented that in CARMA. Do you have an idea how to add vertical BC emissions similar to how we do for surface emissions?

**Comment by zywshoon** [comment]
I don't know that we have BC emissions in Tonga run.

**Comment by tilmes** [comment]
@zywshoon The BC emissions are for aircraft emissions that we have when we run MAM4. I am not sure if those really matter, but it may be useful to also add an option for vertical aerosol emissions.

**Comment by zywshoon** [comment]
Can we ignore it? There's very little BC in the stratosphere. It gets washed out very quickly in the troposphere.
If we want to add it, can't we add the file in the namelist? Is it because we need to put in the CARMA bins?

**Comment by zywshoon** [comment]
I think if we run WACCM MA, it shouldn't matter since we are not even getting tropospheric chemistry.

By the way, if you want to see the changes Chuck made for the ice nucleation. They are in every case we do for Tonga:
SourceMods/src.cam
nucleate_ice_cam.F90  nucleate_ice.F90

**Comment by tilmes** [comment]
@zywshoon the run broke, but you can have an initial look at the output if you have time:
/glade/scratch/tilmes/FWmaCARMAnudged.f09_f09_mg17.carma_trop_strat06.2021_2022/run
let me know if some output is missing, 
Thanks Simone

**Comment by tilmes** [comment]
@zywshoon I did a longer spinup know, starting in 2019 and so the eruption end of 2019 is included. This seems to make a significant difference in the background AOD before the Hunga-Tonga eruption, but not much in March:
<img width="912" alt="image" src="https://user-images.githubusercontent.com/22222754/207415424-ba01cdd3-bf5e-4943-bbb4-80932dcb2d0c.png">
<img width="922" alt="image" src="https://user-images.githubusercontent.com/22222754/207415553-af2d5f9d-91c0-441b-8c82-0e933a134501.png">
<img width="905" alt="image" src="https://user-images.githubusercontent.com/22222754/207415891-e4501a6a-d0f2-4e37-a00b-d87c1b97946d.png">
<img width="921" alt="image" src="https://user-images.githubusercontent.com/22222754/207416071-f4ec507b-0f85-4df2-9f6f-9f659461fa0e.png">

**Comment by zywshoon** [comment]
@tilmes Thank you so much, Simone. able to boost up the background is excellent. I will clone your case next week and see if I get to know more about the sizes and evolutions of the particles.

**Comment by zywshoon** [comment]
<img width="628" alt="Screen Shot 2022-12-16 at 10 50 11 AM" src="https://user-images.githubusercontent.com/77851395/208136607-5d846520-97b1-435b-bc90-d96181910df8.png">

**Comment by zywshoon** [comment]
Hi @tilmes , I checked the Extinction and effective radius. I think our CARMA is producing too large of the particle, while MAM produces too small of particle. So even though the extinction is similar but the reasons are different. That's very interesting. I have a question about the output: in CARMA run output, it is called EXTINCT; in MAM run, it is called EXTINCTdn. Could you confirm there are the same thing? because EXTINCTdn include day-night extinction, while EXTINCT sometimes only account for day time extinction (which may cause some error when we do a monthly average).

Here's the effective radius comparison. The observation should be about 0.4 um. CARMA shows 0.6 um; MAM shows 0.2 um. For a given mass of loading 0.4 um gives the maximum extinction while both 0.6 um and 0.2 um should have lower extinction.

<img width="660" alt="Screen Shot 2022-12-16 at 10 47 56 AM" src="https://user-images.githubusercontent.com/77851395/208137877-8cb1e3be-3d07-4ee2-bcc8-14a1178a4ff0.png">
<img width="625" alt="Screen Shot 2022-12-16 at 10 48 07 AM" src="https://user-images.githubusercontent.com/77851395/208137889-a329a743-056a-4474-9828-de72fa038b54.png">

The extinction comparison:
<img width="669" alt="Screen Shot 2022-12-16 at 10 50 03 AM" src="https://user-images.githubusercontent.com/77851395/208137986-c6e222a6-0d3b-4e6e-8b1c-84a22c4ad534.png">
<img width="628" alt="Screen Shot 2022-12-16 at 10 50 11 AM" src="https://user-images.githubusercontent.com/77851395/208137992-d4060a16-df16-4b33-85bf-f656df2ae394.png">

To summarize and future work:
I will clone your case and run daily output and do an apple-apple comparison.
Why CARMA creates such a large effective radius? Should we enlarger the injection area to reduce the coagulation rate?

**Comment by tilmes** [comment]
@zywshoon We can write out EXTINCT and EXTINCTdn in MAM to make sure there are no differences. If you do another run with CARMA, I can also do a new run with MAM4 to make sure all the emissions are the same. If you point me to the cloned run, I can add the same output. I think, some daily average output may be good to compare for now.

**Comment by zywshoon** [comment]
@tilmes could you point me to the GEOS or MERRA meteorology fields that last until 2023?

**Comment by zywshoon** [comment]
@tilmes I try to print out these O3 variables in the fincl like what we did in MAM runs, but it doesn't allow me. Do you know the reasons?

 FLDLST: OddOx_Ox_Loss in fincl(174, 2) not found
 FLDLST: OddOx_HOx_Loss in fincl(175, 2) not found
 FLDLST: OddOx_NOx_Loss in fincl(176, 2) not found
 FLDLST: OddOx_CLOxBROx_Loss in fincl(177, 2) not found
 FLDLST: OddOx_Loss_Tot in fincl(178, 2) not found
 FLDLST: OddOx_Prod_Tot in fincl(179, 2) not found
 FLDLST: O3_Prod in fincl(180, 2) not found
 FLDLST: O3_Loss in fincl(181, 2) not found
 FLDLST: RO2_NO_sum in fincl(184, 2) not found
 FLDLST: RO2_HO2_sum in fincl(185, 2) not found
 FLDLST: O3S in fincl(186, 2) not found

**Comment by tilmes** [comment]
@zywshoon This is the Middle Atmosphere run, which may not have these diagnostics. I am starting to setup a CAMchem run for 2021-2023. That should have it. I will also now make sure this run has the correct namelist.. stay tuned!

**Comment by tilmes** [comment]
@zywshoon I am spinning up a new model run, based on the latest tag (probably in 2022-07 in 1 day).
/glade/p/cesm/chwg_dev/tilmes/cases/carma/FWmaCARMAnudged.f09_f09_mg17.carma_trop_strat12.2021_2022
output will be in: /glade/scratch/tilmes/archive/FWmaCARMAnudged.f09_f09_mg17.carma_trop_strat12.2021_2022

I am setting up a new run, but using the Vehkamaki nucleation scheme:
/glade/p/cesm/chwg_dev/tilmes/cases/carma/FWmaCARMAnudged.f09_f09_mg17.carma_trop_strat12.2021_2022.vehk
output will be in: /glade/scratch/tilmes/archive/FWmaCARMAnudged.f09_f09_mg17.carma_trop_strat12.2021_2022.vehk

It would be great if you could check if there are any differences compared to the previous version.

**Comment by tilmes** [comment]
@zywshoon 
I just started two more cases (from 2022) that include your new injection run. The output should be there tomorrow evening:
/glade/scratch/tilmes/archive/FWmaCARMAnudged.f09_f09_mg17.carma_trop_strat12.tonga.SO2_0.6Tg
and 
/glade/scratch/tilmes/archive/FWmaCARMAnudged.f09_f09_mg17.carma_trop_strat12.tonga.SO2_0.6Tg.vehk
Using your new file:
SO2_0.6Tg_20220115_Hunga-Tonga_Alt25-35km_AreaTripleRadius250km_1deg_c20230330.nc

**Comment by zywshoon** [comment]
<img width="837" alt="Screen Shot 2023-04-10 at 15 12 36" src="https://user-images.githubusercontent.com/77851395/230999938-1b416f30-d06e-4402-8be5-0b71cfb6aed6.png">

**Comment by tilmes** [comment]
@zywshoon how does this compare to observations?

**Comment by zywshoon** [comment]
better but not good enough.

**Comment by zywshoon** [comment]
Hi @tilmes I'm trying to plot some size distribution but currently the output is monthly average for MXAER01ND, MXAER01WR .... etc.

I tried to clone your case: 
![image](https://github.com/NCAR/CARMA_dev/assets/77851395/20f4541e-39a5-4a32-814c-71df3bcbb554)
But failed when I build: 
ERROR: Command /glade/work/fvitt/camdev/carma_trop_strat06/cime/../bld/build-namelist -ntasks 768 -csmdata /glade/p/cesmdata/cseg/inputdata -infile /glade/u/home/yunqian/cesm/case/CESM2CARMA_Tonga0.6Tg.vehk.daily/Buildconf/camconf/namelist -start_ymd 20220101 -ignore_ic_year -use_case carma_waccm_ma_nudged_cam6 -inputdata /glade/u/home/yunqian/cesm/case/CESM2CARMA_Tonga0.6Tg.vehk.daily/Buildconf/cam.input_data_list -namelist " &atmexp /"  failed rc=255
out=
err=Smartmatch is experimental at /glade/work/fvitt/camdev/carma_trop_strat06/bld/perl5lib/Build/ChemNamelist.pm line 274.
CAM build-namelist - ERROR: Invalid namelist variable in '-infile' /glade/u/home/yunqian/cesm/case/CESM2CARMA_Tonga0.6Tg.vehk.daily/Buildconf/camconf/namelist.
 ERROR: in _validate_pair (package Build::NamelistDefinition): Variable name carma_sulfnuc_method not found in /glade/work/fvitt/camdev/carma_trop_strat06/bld/namelist_files/namelist_definition.xml

I wander if you could rerun the case with daily output for h0 or just add MXAER01ND, MXAER01WR ... to 20 bin, and PURSUL01ND and PULSUL01WR ... to 20 bins to h1 files?

Or you may know what's happening with my build error?
